src/types.ts

```diff
@@ -835,15 +835,17 @@ export class Signature {
   }
 
   /** Tests if a value of this function type is assignable to a target of the specified function type. */
-  isAssignableTo(target: Signature, requireSameSize: bool = false): bool {
-
-    // check `this` type
-    var thisThisType = this.thisType;
-    var targetThisType = target.thisType;
-    if (thisThisType) {
-      if (!targetThisType || !thisThisType.isAssignableTo(targetThisType)) return false;
-    } else if (targetThisType) {
-      return false;
+  isAssignableTo(target: Signature, requireSameSize: bool = false, isInterface: bool = false): bool {
+
+    if (!isInterface) {
+      // check `this` type
+      var thisThisType = this.thisType;
+      var targetThisType = target.thisType;
+      if (thisThisType) {
+        if (!targetThisType || !thisThisType.isAssignableTo(targetThisType)) return false;
+      } else if (targetThisType) {
+        return false;
+      }
```

Comment:
One case I could imagine where skipping the check entirely is an override of `(this: i32, ...): ...`, where then there would be two classes extending the same interface with an incompatible `this` type. Perhaps keep a check for managed-ness / strict assignability of the types?